plugins/modem-manager/fu-plugin-modem-manager.c

```diff
@@ -60,17 +60,20 @@ static void
 fu_plugin_mm_uninhibit_device(FuPlugin *plugin)
 {
 	FuPluginData *priv = fu_plugin_get_data(plugin);
-	g_autoptr(FuPluginMmInhibitedDeviceInfo) info = NULL;
 
+	/* Udev monitoring can be removed as modem is back again */
 	g_clear_object(&priv->udev_client);
 
 	/* get the device removed from the plugin cache before uninhibiting */
 	fu_plugin_mm_udev_device_removed(plugin);
 
-	info = g_steal_pointer(&priv->inhibited);
-	if ((priv->manager != NULL) && (info != NULL)) {
-		g_debug("uninhibit modemmanager device with uid %s", info->inhibited_uid);
-		mm_manager_uninhibit_device_sync(priv->manager, info->inhibited_uid, NULL, NULL);
+	if ((priv->manager != NULL) && (priv->inhibited != NULL)) {
+		g_debug("uninhibit modemmanager device with uid %s",
+			priv->inhibited->inhibited_uid);
+		mm_manager_uninhibit_device_sync(priv->manager,
+						 priv->inhibited->inhibited_uid,
+						 NULL,
+						 NULL);
 	}
```

Comment:
I don't think this is right; the original code cleaned up priv->inhibited.

Reply:
Hmmmm that's true because we don't do `g_steal_pointer` anymore.
What about cleaning it up in `fu_plugin_mm_inhibit_device`?
Right before we do `g_steal_pointer` there, free the value if there is one?

```
diff --git a/plugins/modem-manager/fu-plugin-modem-manager.c b/plugins/modem-manager/fu-plugin-modem-manager.c
index 73f5aa44..f4185a63 100644
--- a/plugins/modem-manager/fu-plugin-modem-manager.c
+++ b/plugins/modem-manager/fu-plugin-modem-manager.c
@@ -203,6 +203,7 @@ fu_plugin_mm_inhibit_device(FuPlugin *plugin, FuDevice *device, GError **error)
                return FALSE;
 
        /* setup inhibited device info */
+       g_free(priv->inhibited);
        priv->inhibited = g_steal_pointer(&info);
 
        /* only do modem port monitoring using udev if the module is expected
```

Reply:
`priv->inhibited` isn't a single pointer; it's a `FuPluginMmInhibitedDeviceInfo` object, hence the `g_autoptr`. I think the code needs to change back to what it was before tbh.

Reply:
I reverted the change, retried it a few times and it seems to be fine now. Maybe it got fixed somehow with other commits.